a4_APIweatherApp.py

```diff
@@ -0,0 +1,90 @@
+'''
+This program is inteded to use the Open Meteo API
+and output an API and Mock plot of temperatures in the next 24 hours
+'''
+
+from abc import ABC, abstractmethod
+import requests
+import matplotlib.pyplot as plt
+
+class Service(ABC):
+    @abstractmethod
+    def getting(self):
+        pass
+
+class mockService(Service):
+    def getting(self):
+        #create a 
+        dates = ["2024-08-02T02:00",
+      "2024-08-02T03:00",
+      "2024-08-02T04:00",
+      "2024-08-02T05:00",
+      "2024-08-02T06:00",
+      "2024-08-02T07:00",
+      "2024-08-02T08:00",
+      "2024-08-02T09:00",
+      "2024-08-02T10:00",
+      "2024-08-02T11:00",
+      "2024-08-02T12:00",
+      "2024-08-02T13:00",
+      "2024-08-02T14:00",
+      "2024-08-02T15:00",
+      "2024-08-02T16:00",
+      "2024-08-02T17:00",
+      "2024-08-02T18:00",
+      "2024-08-02T19:00",
+      "2024-08-02T20:00",
+      "2024-08-02T21:00",
+      "2024-08-02T22:00",
+      "2024-08-02T23:00",
+      "2024-08-03T00:00",
+      "2024-08-03T01:00"]
+        temps = [19.4, 18.8, 18.2, 17.8, 17.7, 17.7, 17.9, 18, 18.1, 18.6, 18.8, 18.8, 18.9, 19.2, 19.4, 19.6, 19.4, 19.5, 19.4, 19, 18.5, 18, 17.6, 17.3]
+        plt.scatter(dates, temps)
+        plt.title("Mock Service")
+        plt.show()
+
+
+class APIService(Service):
+    def getting(self):
+        response = requests.get("https://api.open-meteo.com/v1/forecast?latitude=52.52&longitude=13.41&hourly=temperature_2m")
+        dates = response.json()['hourly']['time'][0:24]
+        temps = response.json()['hourly']['temperature_2m'][0:24]
+        plt.scatter(dates, temps)
+        plt.title("API Service")
+        plt.show()
+
+
+class ServiceFactory(ABC):
+    def create_service(self, num):
+        #if num = 1, then APIService, if num = 0, then mockService
```

Comment:
When you are looking for this pattern (some ints defining some state that you care about), instead of having what's called 'magic numbers' all over your code, try using Enums. For this example you can write something like

```
from enum import Enum

class Environment(Enum):
    TEST = 0
    PROD = 1
```

then, you can re-write to expect `create_service` an enum, and you can write your conditionals in a way that makes for semantic sense about the context of your code

```
if num == Environment.PROD:
    return APIService()
elif num = Environment.TEST:
    return MockSerivce
```

This would also allow you to rename the `num` variable to something that makes more sense in this context, and it not a generic name. Maybe something like `environment` or `execution_env`?

Another benefit here is that if you ever have to add/remove the number of environments your support, or change the order, you can just do that in the definition of the Enum, and be sure that won't break anything in your implementation.